interfaces/cython/cantera/thermo.pyx

```diff
@@ -2045,6 +2053,18 @@ cdef class PureFluid(ThermoPhase):
         def __get__(self):
             return self.s, self.v, self.Q
 
+# TODO: Remove these helper methods when support for Python 3.8 is dropped. Python 3.9
+# allows the classmethod and property decorators to be chained, so these can be
+# implemented as properties in the Element class.
```

Comment:
Wouldn't making these into class properties mean that this tuple was constructed again from the C++ `elementSymbols` method every time it was accessed, rather than just a single time as a class attribute?

Reply:
For me, trying the above snippet in Python 3.10 gives the error: `TypeError: element_symbols() takes no arguments (1 given)`. Is there something else I'm missing?

Reply:
Right, I should have known that. On my system, `ct.Element.element_symbols is ct.Element.element_symbols` returns `False` (executing from within IPython). And putting a `print` into the method shows that it is being executed every time.

Reply:
Ack, this is what I get for explaining from memory. I also added a class variable `_element_symbols = None`, so it looks like:

```python
class Element:
...
    _element_symbols = None
...
    @classmethod
    @property
    def element_symbols(cls):
        if cls._element_symbols is None:
            syms = elementSymbols()
            cls._element_symbols = tuple(pystr(s) for s in syms)
        return cls._element_symbols
```

By the way, I'd prefer to do `element_symbols = tuple(pystr(s) for s in elementSymbols())` but that gives me a compiler error, something about `lvalue` or `rvalue` not being able to be used as a reference. By experimenting, I interpreted that to mean that since `elementSymbols()` is used in the comprehension, it gets assigned to a reference by Cython in `_cantera.cpp`, and if I assign it to a variable, things are better.

Reply:
Ah, I see. Does updating `cantera.pxd` to indicate that `elementSymbols` returns a `vector[string]&` get Cython to generate something that actually compiles?

Reply:
No, then I get:

```
clang++ -o build/temp-py/_cantera.os -c -std=c++11 -fcolor-diagnostics -pthread -O0 -g -fPIC -Iinclude -I/home/bryanw/mambaforge/envs/cantera-dev/include/python3.10 -I/home/bryanw/mambaforge/envs/cantera-dev/lib/python3.10/site-packages/numpy/core/include -I/home/bryanw/mambaforge/envs/cantera-dev/include build/python/cantera/_cantera.cpp
...
build/python/cantera/_cantera.cpp:63552:15: error: assigning to 'std::vector<std::string> *' (aka 'vector<basic_string<char>> *') from 'const std::vector<std::string> *' (aka 'const vector<basic_string<char>> *') discards qualifiers
  __pyx_t_2 = &Cantera::elementSymbols();
              ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Ah, too bad. I tried a couple of other things but couldn't come up with one that made the one-liner version of this actually work.

Reply:
Yeah, I tried looking around the internet and couldn't find a good answer either. Darn.